ion/integration_test.go

```diff
@@ -0,0 +1,807 @@
+			}
+			err = writer.WriteInt(val)
+			if err != nil {
+				t.Errorf("Something went wrong when writing Int value. " + err.Error())
+			}
+
+		case FloatType:
+			val, err := reader.FloatValue()
+			if err != nil {
+				t.Errorf("Something went wrong when reading Float value. " + err.Error())
+			}
+			err = writer.WriteFloat(val)
+			if err != nil {
+				t.Errorf("Something went wrong when writing Float value. " + err.Error())
+			}
+
+		case DecimalType:
+			val, err := reader.DecimalValue()
+			if err != nil {
+				t.Errorf("Something went wrong when reading Decimal value. " + err.Error())
+			}
+			err = writer.WriteDecimal(val)
+			if err != nil {
+				t.Errorf("Something went wrong when writing Decimal value. " + err.Error())
+			}
+
+		case TimestampType:
+			val, err := reader.TimeValue()
+			if err != nil {
+				t.Errorf("Something went wrong when reading Timestamp value. " + err.Error())
+			}
+			err = writer.WriteTimestamp(val)
+			if err != nil {
+				t.Errorf("Something went wrong when writing Timestamp value. " + err.Error())
+			}
+
+		case SymbolType:
+			val, err := reader.StringValue()
+			if err != nil {
+				t.Errorf("Something went wrong when reading Symbol value. " + err.Error())
+			}
+			err = writer.WriteSymbol(val)
+			if err != nil {
+				t.Errorf("Something went wrong when writing Symbol value. " + err.Error())
+			}
+
+		case StringType:
+			val, err := reader.StringValue()
+			if err != nil {
+				t.Errorf("Something went wrong when reading String value. " + err.Error())
+			}
+			err = writer.WriteString(val)
+			if err != nil {
+				t.Errorf("Something went wrong when writing String value. " + err.Error())
+			}
+
+		case ClobType:
+			val, err := reader.ByteValue()
+			if err != nil {
+				t.Errorf("Something went wrong when reading Clob value. " + err.Error())
+			}
+			err = writer.WriteClob(val)
+			if err != nil {
+				t.Errorf("Something went wrong when writing Clob value. " + err.Error())
+			}
+
+		case BlobType:
+			val, err := reader.ByteValue()
+			if err != nil {
+				t.Errorf("Something went wrong when reading Blob value. " + err.Error())
+			}
+			err = writer.WriteBlob(val)
+			if err != nil {
+				t.Errorf("Something went wrong when writing Blob value. " + err.Error())
+			}
+
+		case SexpType:
+			reader.StepIn()
+			writer.BeginSexp()
+			writeToWriterFromReader(t, reader, writer)
+			reader.StepOut()
+			writer.EndSexp()
+
+		case ListType:
+			reader.StepIn()
+			writer.BeginList()
+			writeToWriterFromReader(t, reader, writer)
+			reader.StepOut()
+			writer.EndList()
+
+		case StructType:
+			reader.StepIn()
+			writer.BeginStruct()
+			writeToWriterFromReader(t, reader, writer)
+			reader.StepOut()
+			writer.EndStruct()
+		}
+	}
+
+	if reader.Err() != nil {
+		t.Errorf(reader.Err().Error())
+	}
+}
+
+// Read the current value in the reader and put that in an item struct (defined in this file).
+func readCurrentValue(t *testing.T, reader Reader) item {
+	var item item
```

Comment:
Without extra context, this variable declaration was a bit puzzling. Maybe we could rename the internal struct type `ionItem` or `ionValue` for clarity?

Reply:
Renamed to ```ionItem```. Thanks.